src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1802,7 +1802,7 @@ private int GetNextCaptureId(RegionBuilder owner)
             return captureId;
         }
 
-        private void SpillEvalStack()
+        private void SpillEvalStack(bool spillDeclarationExpressions = false)
```

Comment:
It feels like we might spill too much. I.e. spill `DeclarationExpression`s that do not belong to the immediate explicit call. We probably should limit this exception by ```_currentInterpolatedStringHandlerArgumentContext.StartingStackDepth```, but then we also might need to drop the optimization based on ```_startSpillingAt```.

Reply:
On the second thought, this optimization will come at the cost of figuring out the right spill region for the spilling "out of order". Not worth the risk and complexity, I think.